### PR TITLE
Update to Ruby 2.6.2 and Rubygems 3.0.3

### PR DESCRIPTION
**Pull request by tas50**
We thought we were riding the latest, but actually the overrides file wasn't wired up the same way it is in Chef. This properly sources that file and does a few other syncs from the chef config so we can diff things more easily.

This resolves:

- CVE-2019-8320: Delete directory using symlink when decompressing tar
- CVE-2019-8321: Escape sequence injection vulnerability in verbose
- CVE-2019-8322: Escape sequence injection vulnerability in gem owner
- CVE-2019-8323: Escape sequence injection vulnerability in API response handling
- CVE-2019-8324: Installing a malicious gem may lead to arbitrary code execution
- CVE-2019-8325: Escape sequence injection vulnerability in errors
- Prevent path traversal when writing to a symlinked basedir outside of the root. Discovered by nmalkin and David Fifield, fixed by Jonathan Claudius and Samuel Giddins.
- Fix possible Unsafe Object Deserialization Vulnerability in gem owner. Fixed by Jonathan Claudius.
- Strictly interpret octal fields in tar headers. Discoved by plover, fixed by Samuel Giddins.
- Raise a security error when there are duplicate files in a package. Discovered by plover, fixed by Samuel Giddins.
- Enforce URL validation on spec homepage attribute. Discovered by Yasin Soliman, fixed by Jonathan Claudius.
- Mitigate XSS vulnerability in homepage attribute when displayed via gem server. Discovered by Yasin Soliman, fixed by Jonathan Claudius.
- Prevent Path Traversal issue during gem installation. Discovered by nmalkin and David Fifield.

Signed-off-by: Tim Smith <tsmith@chef.io>

### CODE REVIEW

**Review by miah — APPROVED** [review]

File: omnibus/Gemfile [hunk 1]
```diff
@@ -23,7 +13,13 @@ group :development do
   gem 'berkshelf', '>= 7.0'
 
   # Use Test Kitchen with Vagrant for converging the build environment
-  gem 'kitchen-vagrant', '~> 1.3.1'
-  gem 'test-kitchen',    '~> 1.21'
+  gem 'kitchen-vagrant', '>= 1.3.1'
+  gem 'test-kitchen', '>= 1.23'
```
Comment (tas50) [inline omnibus/Gemfile:17]: This allows us to start using test-kitchen 2.0 which we want for licensing support

**Review by lamont-granquist — APPROVED** [review]

File: omnibus/Gemfile [hunk 2]
```diff
@@ -23,7 +13,13 @@ group :development do
...
   gem 'winrm-fs', '~> 1.0'
 end
+
+instance_eval(ENV['GEMFILE_MOD']) if ENV['GEMFILE_MOD']
```
Comment (zenspider) [inline omnibus/Gemfile:21]: This seems odd to me. Is this supposed to be an arbitrary ruby string or a path to a file to eval in?

File: omnibus_overrides.rb [hunk 3]
```diff
@@ -1 +1,12 @@
-# this file is used to override versions of packages in the omnibus build
+# THIS IS NOW HAND MANAGED, JUST EDIT THE THING
```
Comment (zenspider) [inline omnibus_overrides.rb:1]: I :heart: this line.

File: omnibus/config/projects/inspec.rb [hunk 4]
```diff
@@ -37,14 +37,9 @@
...
-# RubyGems 2.7.0 caused issues in the Jenkins pipelines, trouble installing bundler.
-# This issue is not evident in 2.6.x, hence the pin.
-override 'rubygems', version: '2.6.14'
-
-# grab the current train release from rubygems.org
-train_stable = /^train \((.*)\)/.match(`gem list ^train$ --remote`)[1]
-override 'train', version: "v#{train_stable}"
+# Load dynamically updated overrides
+overrides_path = File.expand_path('../../../../omnibus_overrides.rb', __FILE__)
+instance_eval(IO.read(overrides_path), overrides_path)
```
Comment (zenspider) [inline omnibus/config/projects/inspec.rb]: Please use `File.read`

File: omnibus/config/projects/inspec.rb [hunk 5]
```diff
@@ -58,10 +58,8 @@
 
 package :rpm do
   signing_passphrase ENV['OMNIBUS_RPM_SIGNING_PASSPHRASE']
-  unless rhel? && platform_version.satisfies?('< 6')
-    compression_level 1
-    compression_type :xz
-  end
+  compression_level 1
+  compression_type :xz
```
Comment (zenspider) [inline omnibus/config/projects/inspec.rb:62]: Should this be a separate commit? What's this about?

**Review by zenspider — DISMISSED** [review]

File: omnibus/config/projects/inspec.rb (hunk 5 above)
Comment (tas50) [reply omnibus/config/projects/inspec.rb:62]: We haven't built RHEL 5 packages for a while. This is a leftover from the olden days of our omnibus config.

File: omnibus/Gemfile (hunk 2 above)
Comment (tas50) [reply omnibus/Gemfile:21]: @lamont-granquist how were you using this locally?

**Comment by clintoncwolfe** [comment]
I'm going to go ahead and merge this so we can ship 4

**Pull request merged**